ibis/backends/dask/execution/util.py

```diff
@@ -345,3 +348,116 @@ def safe_scalar_type(output_meta):
         output_meta = pd.Timestamp(1, tz=output_meta.tz, unit=output_meta.unit)
 
     return output_meta
+
+
+def add_partitioned_sorted_column(
+    df: Union[dd.DataFrame, dd.Series],
+    set_index: bool = False,
+    col_name: str = "_ibis_index",
+) -> Tuple[dd.DataFrame, List]:
+    """Add a column that is already partitioned and sorted
+    This columns acts as if we had a global index across the distributed data.
+    Important properties:
+    - Each row has a unique id (i.e. a value in this column)
+    - IDs within each partition are already sorted
+    - Any id in partition N_{t} is less than any id in partition N_{t+1}
+    We do this by designating a sufficiently large space of integers per
+    partition via a base and adding the existing index to that base. See
+    `helper` below.
+    Though the space per partition is bounded, real world usage should not
+    hit these bounds. We also do not explicity deal with overflow in the
+    bounds.
+
+    Parameters
+    ----------
+    df : dd.DataFrame
+        Dataframe to add a column to
+    set_index : bool, optional
+        Whether to also set that column as the index, by default False
+    col_name : str, optional
+        The column name, by default "_ibis_index"
```

Comment:
Maybe I am not understanding this correctly, but how do you communicate the column name from the producer of the index column and the consumer of the index column? (e.g., if the producer creates a index column "a", how would the downstream consumer know that "a" is an index column?)

Reply:
Maybe `_row_index` is a better name than `_ibis_index`

Reply:
Or just `_index`

Reply:
> Maybe I am not understanding this correctly, but how do you communicate the column name from the producer of the index column and the consumer of the index column? 

Convention inside the ibis dask backend. This could have been hardcoded, but it felt easy enough to make it configurable.

> e.g., if the producer creates a index column "a", how would the downstream consumer know that "a" is an index column?

Theoretically we could have made it a global or hardcoded it, but nothing currently relies on this name explicitly. I think I will need to refer to this column explicitly when I do the grouped window udfs, but haven't gotten there yet. 

I have no real view on the name, but given that it is only used internally in ibis inside the dask backend, I'm happy to change it to anything you and Jeff agree on.

Reply:
> I think I will need to refer to this column explicitly when I do the grouped window udfs, but haven't gotten there yet.
It's a little hard for me to understand what you plan to do with different names with grouped window udfs so commenting sth like

"the names are technically configurable but not used yet. It might be used in the future so leaving it configurable for now."

So a bit more clear to reviewer of this PR

Reply:
Apologies for the confusion - I've just dropped this configuration for now.